### PR TITLE
MM-19755 Add landmark roles

### PR DESCRIPTION
**Pull request by nadalfederer**
#### Summary
Add landmark roles
#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/13367
Screenshot from 2019-12-15 19-05-18

### CODE REVIEW

File: components/post_view/post_view.jsx [hunk 1]
```diff
@@ -67,7 +67,10 @@ export default class PostView extends React.PureComponent {
         }
 
         return (
-            <div id='post-list'>
+            <div
+                id='post-list'
+                role='form'
```
Comment (saturninoabril) [inline components/post_view/post_view.jsx]: Is this the one for "The post component (on both center and RHS) could be role='form'"? I thought it's the textbox on center and RHS but others can confirm.

**Comment by hmhealey** [comment]
Deferring to Devin on this review since he's got a better understanding of the accessibility requirements

File: components/post_view/post_view.jsx (hunk 1 above)
Comment (devinbinnie) [reply components/post_view/post_view.jsx]: Yes, that's correct. This should be `role='main'`

File: components/sidebar/sidebar.jsx [hunk 2]
```diff
@@ -754,6 +754,7 @@ export default class Sidebar extends React.PureComponent {
                 className={classNames('sidebar--left', {'move--right': this.props.isOpen && Utils.isMobile()})}
                 id='sidebar-left'
                 key='sidebar-left'
+                role='main'
```
[the remark this answers is not in the document]
Comment (devinbinnie) [reply components/sidebar/sidebar.jsx]: Yes, this should be `role='navigation'`

[the remark this answers is not in the document]
Comment (devinbinnie) [reply components/sidebar/sidebar.jsx]: Yeah, this one should be `role='navigation'`

File: components/team_sidebar/team_sidebar_controller.jsx [hunk 3]
```diff
@@ -139,6 +139,7 @@ export default class TeamSidebar extends React.PureComponent {
                 <div
                     className='team-wrapper'
                     id='teamSidebarWrapper'
+                    role='navigation'
```
Comment (devinbinnie) [inline components/team_sidebar/team_sidebar_controller.jsx]: Can we remove the role here in favour of `sidebar.jsx`?

File: components/create_post/create_post.jsx [hunk 4]
```diff
@@ -1314,6 +1314,7 @@ export default class CreatePost extends React.Component {
                     </div>
                     <div
                         id='postCreateFooter'
+                        role='contentinfo'
```
Comment (devinbinnie) [inline components/create_post/create_post.jsx]: This should be `role='form'`

Comment (devinbinnie) [reply components/create_post/create_post.jsx]: We also need to add that role to the `create_comment` component.

File: components/team_sidebar/team_sidebar_controller.jsx (hunk 3 above)
Comment (nadalfederer) [reply components/team_sidebar/team_sidebar_controller.jsx]: that will meant that this part is visible and not under any role and `axe` will return an error
Screenshot from 2019-12-20 20-03-37
What am I doing wrong?

Comment (devinbinnie) [reply components/team_sidebar/team_sidebar_controller.jsx]: Does having `role='navigation'` on both `sidebar` and `team_sidebar_controller` work?

Comment (nadalfederer) [reply components/team_sidebar/team_sidebar_controller.jsx]: It's like this now.` components/sidebar/sidebar.jsx ` and `components/team_sidebar/team_sidebar_controller.jsx ` have `role='navigation'`. There is no error like this from axe

Comment (devinbinnie) [reply components/team_sidebar/team_sidebar_controller.jsx]: Okay, we're probably covered then.

File: components/team_sidebar/team_sidebar_controller.jsx [hunk 5]
```diff
@@ -135,7 +135,10 @@ export default class TeamSidebar extends React.PureComponent {
         );
 
         return (
-            <div className={classNames('team-sidebar', {'move--right': this.props.isOpen})}>
+            <div
+                className={classNames('team-sidebar', {'move--right': this.props.isOpen})}
+                role='navigation'
```
Comment (saturninoabril) [inline components/team_sidebar/team_sidebar_controller.jsx]: With axe, I've got error of "Ensures landmarks are unique". Suggested fix is "The landmark must have a unique aria-label, aria-labelledby, or title to make landmarks distinguishable"

Screen Shot 2020-01-07 at 4 34 04 PM

**Comment by hanzei** [comment]
@gsagula @saturninoabril Would you please look into this PR when you are back from you weekend break? Thank you :wink:

**Comment by jasonblais** [comment]
Hi @gsagula, reminder to help with a PR review when you have a moment?

**Comment by nadalfederer** [comment]
Hi, is there something I have to do for this PR to be reviewed?

**Comment by jasonblais** [comment]
@gsagula please review when you have a moment, or delegate to someone else? Thanks!

**Comment by jasonblais** [comment]
cc @hanzei

**Comment by gsagula** [comment]
@hanzei Sorry for the delay on this. It looks like that this PR has only changes to the React front-end app. I can do my best but I'm not sure if I'm the most qualified person to review it.

**Comment by nadalfederer** [comment]
test failed after the update branch, I don't know why. I'll fix it

**Comment by Willyfrog** [comment]
can you please update the snapshots so the tests don't fail?

**Comment by Willyfrog** [comment]
thank you @nadalfederer

**Pull request merged**